go/staking/api/commission.go

```diff
@@ -0,0 +1,269 @@
+package api
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/oasislabs/oasis-core/go/common/quantity"
+	epochtime "github.com/oasislabs/oasis-core/go/epochtime/api"
+)
+
+// CommissionRateDenominator is the denominator for the commission rate.
+var CommissionRateDenominator *quantity.Quantity
+
+type CommissionRateStep struct {
+	Start epochtime.EpochTime `json:"start"`
+	Rate  quantity.Quantity   `json:"rate"`
+}
+
+type CommissionRateBoundStep struct {
+	Start   epochtime.EpochTime `json:"start"`
+	RateMin quantity.Quantity   `json:"rate_min"`
+	RateMax quantity.Quantity   `json:"rate_max"`
+}
+
+type CommissionSchedule struct {
+	Rates  []CommissionRateStep      `json:"rates"`
+	Bounds []CommissionRateBoundStep `json:"bounds"`
+}
+
+// validateNondegenerate detects degenerate steps.
+func (cs *CommissionSchedule) validateNondegenerate(commissionRateChangeInterval epochtime.EpochTime) error {
```

Comment:
We should probably limit the maximum number of steps. An implicit limit is the transaction size, but maybe we should choose a sensible higher-level limit as well.

Reply:
ok I'll add that